### ISSUE TITLE
Build error with gcc 7.5.0, mochi-bedrock

### ISSUE THREAD

**Issue by dyokelson** [mochi-hpc/mochi-thallium#10]
I'm getting the following error during the 'spack install' step of a project based on the thallium template. This error occurs during the build phase of mochi-bedrock. 
Using gcc 7.5.0, spack 19, on AMD EPYC 7742
Please let me know if you need any additional information

```
  thallium/engine.hpp:1173:8: sorry, unimplemented: non-trivial designated initializers not supported
  71   inline __MARGO_INTERNAL_RPC_WRAPPER(thallium_generic_rpc)
  72       ^
  73   make[2]: *** [src/CMakeFiles/bedrock-client.dir/build.make:107: src/CMakeFiles/bedrock-
      client.dir/AsyncRequest.cpp.o] Error 1
```

**Comment by mdorier** [comment]
Can you tell me the version of margo you are using?

**Comment by dyokelson** [comment]
Thallium is 0.10.1, but I don't specify the version in spack.yaml, so I think it should just pull the newest. I will update spack.yaml to explicitly choose it. 
It looks like it installed margo 0.13 during the spack install, but that isn't in the spack.yaml at all, would it help to explicitly choose a version for this too?

**Comment by mdorier** [comment]
Thallium 0.10.1 is not the latest one (we are at 0.11.0 now). The error is strange, when I google about it it shows up for much older versions of gcc. Are you 100% sure spack is using gcc 7.5.0? By any chance do you have a newer gcc to test with?

**Comment by dyokelson** [comment]
Oh ok I will specify 0.11.0 instead for thallium. 
Here is an example output line from 'spack install' 
`mochi-thallium@0.10.1%gcc@7.5.0+cereal~ipo build_system=cmake build_type=RelWithDebInfo arch=linux-sles15-zen`
When I run 'spack compiler find' the only available gcc is 7.5.0. I did successfully run 'spack install' when I used gcc 11.2.0 on this machine, the problem is for the cray programming environment set up by the sys admins on this machine when I load a different gcc it replaces the cce compiler that is then required when I try to run 'make install' for the template code. The other compiler options in this programming environment are cce 14 and clang 14. I will ask the sys admins to upgrade gcc (I don't know why it's such an old version anyway). I have successfully built with gcc 9.4.0 on a different machine as well.

**Comment by mdorier** [comment]
Why use cce when compiling the template code? If that's because it's the one cmake picks, you can change that:
```
cmake ... -DCMAKE_CXX_COMPILER=g++ -DCMAKE_C_COMPILER=gcc
```
Besides, if your spack-installed stack is built with gcc, you should probably build the template with gcc as well.

**Comment by dyokelson** [comment]
Ah, right now I use flag `-DCMAKE_CXX_COMPILER=mpicxx`
which I thought looked for cce in order to get mpich for MPI, perhaps I'm mistaken and don't need cce for MPI functionality. I can try that out. 
However, I also tried getting spack to install the dependencies using cce 14 and clang 14, but in both cases there were 10 or more build failures during 'spack install' all for different dependencies (many looked like failed recipes not related to any mochi technology)...I may ask the spack team if there is something going on there.

**Comment by dyokelson** [comment]
Here is more detailed error output in case it's helpful. I've also requested an upgraded gcc on the machine for this programming environment. 

-- Build files have been written to: /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o
==> mochi-bedrock: Executing phase: 'build'
==> [2023-03-22-00:24:45.666604] 'make' '-j16'
/pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/cmake-3.24.3-vkdicyfnbvvdbhghumodnsw43jty6zob/bin/cmake -S/tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-src -B/tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o --check-build-system CMakeFiles/Makefile.cmake 0
/pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/cmake-3.24.3-vkdicyfnbvvdbhghumodnsw43jty6zob/bin/cmake -E cmake_progress_start /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o/CMakeFiles /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o//CMakeFiles/progress.marks
make  -f CMakeFiles/Makefile2 all
make[1]: Entering directory '/tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o'
make  -f src/CMakeFiles/jx9.dir/build.make src/CMakeFiles/jx9.dir/depend
make  -f src/CMakeFiles/bedrock-client.dir/build.make src/CMakeFiles/bedrock-client.dir/depend
make[2]: Entering directory '/tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o'
cd /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o && /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/cmake-3.24.3-vkdicyfnbvvdbhghumodnsw43jty6zob/bin/cmake -E cmake_depends "Unix Makefiles" /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-src /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-src/src /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o/src /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o/src/CMakeFiles/jx9.dir/DependInfo.cmake --color=
make[2]: Entering directory '/tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o'
cd /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o && /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/cmake-3.24.3-vkdicyfnbvvdbhghumodnsw43jty6zob/bin/cmake -E cmake_depends "Unix Makefiles" /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-src /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-src/src /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o/src /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o/src/CMakeFiles/bedrock-client.dir/DependInfo.cmake --color=
make[2]: Leaving directory '/tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o'
make  -f src/CMakeFiles/jx9.dir/build.make src/CMakeFiles/jx9.dir/build
make[2]: Leaving directory '/tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o'
make  -f src/CMakeFiles/bedrock-client.dir/build.make src/CMakeFiles/bedrock-client.dir/build
make[2]: Entering directory '/tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o'
make[2]: Entering directory '/tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o'
[  4%] Building C object src/CMakeFiles/jx9.dir/jx9/jx9.c.o
cd /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o/src && /pfs/lustrep1/users/dewiy/spack/lib/spack/env/gcc/gcc -DENABLE_ABT_IO -DENABLE_SSG  -O2 -g -DNDEBUG -fPIC   -g -MD -MT src/CMakeFiles/jx9.dir/jx9/jx9.c.o -MF CMakeFiles/jx9.dir/jx9/jx9.c.o.d -o CMakeFiles/jx9.dir/jx9/jx9.c.o -c /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-src/src/jx9/jx9.c
[  8%] Building CXX object src/CMakeFiles/bedrock-client.dir/Client.cpp.o
[ 12%] Building CXX object src/CMakeFiles/bedrock-client.dir/ServiceHandle.cpp.o
cd /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o/src && /pfs/lustrep1/users/dewiy/spack/lib/spack/env/gcc/g++ -DENABLE_ABT_IO -DENABLE_SSG -DFMT_SHARED -DSPDLOG_COMPILED_LIB -DSPDLOG_SHARED_LIB -Dbedrock_client_EXPORTS -I/tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o/src -I/tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-src/src/../include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/mochi-thallium-0.10.1-ux2i6ha3jhuw5b3tcuhgrqtwvz374mun/include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/mochi-margo-0.13-lw73m7isbhdvm3s6vcipckxgoqkiglp3/include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/mercury-2.2.0-kaljvckxgvcznepx57drwangvhoee5dk/include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/boost-1.80.0-nguzaqdg2ugrxqzsblvppmk24whp5vwd/include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/json-c-0.16-jk2gtg7cxqfofna22acedbp7q2jvtdzc/include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/json-c-0.16-jk2gtg7cxqfofna22acedbp7q2jvtdzc/include/json-c -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/argobots-1.1-ofl32sm75srh3o4qgtnwei3mpfuigcz4/include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/cereal-1.3.2-5q63tts5i7ijmha4k5a6lpzmpd5yr3xg/include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/spdlog-1.10.0-xvgujvylicokkeotqfa5ae72cvz26bgi/include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/fmt-9.1.0-wzitn7apw7mtivdkrb4udfu6fmtdzdu6/include -O2 -g -DNDEBUG -fPIC   -g -Wall -Wextra -pthread -std=gnu++14 -MD -MT src/CMakeFiles/bedrock-client.dir/Client.cpp.o -MF CMakeFiles/bedrock-client.dir/Client.cpp.o.d -o CMakeFiles/bedrock-client.dir/Client.cpp.o -c /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-src/src/Client.cpp
cd /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o/src && /pfs/lustrep1/users/dewiy/spack/lib/spack/env/gcc/g++ -DENABLE_ABT_IO -DENABLE_SSG -DFMT_SHARED -DSPDLOG_COMPILED_LIB -DSPDLOG_SHARED_LIB -Dbedrock_client_EXPORTS -I/tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o/src -I/tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-src/src/../include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/mochi-thallium-0.10.1-ux2i6ha3jhuw5b3tcuhgrqtwvz374mun/include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/mochi-margo-0.13-lw73m7isbhdvm3s6vcipckxgoqkiglp3/include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/mercury-2.2.0-kaljvckxgvcznepx57drwangvhoee5dk/include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/boost-1.80.0-nguzaqdg2ugrxqzsblvppmk24whp5vwd/include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/json-c-0.16-jk2gtg7cxqfofna22acedbp7q2jvtdzc/include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/json-c-0.16-jk2gtg7cxqfofna22acedbp7q2jvtdzc/include/json-c -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/argobots-1.1-ofl32sm75srh3o4qgtnwei3mpfuigcz4/include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/cereal-1.3.2-5q63tts5i7ijmha4k5a6lpzmpd5yr3xg/include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/spdlog-1.10.0-xvgujvylicokkeotqfa5ae72cvz26bgi/include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/fmt-9.1.0-wzitn7apw7mtivdkrb4udfu6fmtdzdu6/include -O2 -g -DNDEBUG -fPIC   -g -Wall -Wextra -pthread -std=gnu++14 -MD -MT src/CMakeFiles/bedrock-client.dir/ServiceHandle.cpp.o -MF CMakeFiles/bedrock-client.dir/ServiceHandle.cpp.o.d -o CMakeFiles/bedrock-client.dir/ServiceHandle.cpp.o -c /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-src/src/ServiceHandle.cpp
[ 16%] Building CXX object src/CMakeFiles/bedrock-client.dir/AsyncRequest.cpp.o
cd /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o/src && /pfs/lustrep1/users/dewiy/spack/lib/spack/env/gcc/g++ -DENABLE_ABT_IO -DENABLE_SSG -DFMT_SHARED -DSPDLOG_COMPILED_LIB -DSPDLOG_SHARED_LIB -Dbedrock_client_EXPORTS -I/tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o/src -I/tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-src/src/../include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/mochi-thallium-0.10.1-ux2i6ha3jhuw5b3tcuhgrqtwvz374mun/include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/mochi-margo-0.13-lw73m7isbhdvm3s6vcipckxgoqkiglp3/include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/mercury-2.2.0-kaljvckxgvcznepx57drwangvhoee5dk/include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/boost-1.80.0-nguzaqdg2ugrxqzsblvppmk24whp5vwd/include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/json-c-0.16-jk2gtg7cxqfofna22acedbp7q2jvtdzc/include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/json-c-0.16-jk2gtg7cxqfofna22acedbp7q2jvtdzc/include/json-c -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/argobots-1.1-ofl32sm75srh3o4qgtnwei3mpfuigcz4/include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/cereal-1.3.2-5q63tts5i7ijmha4k5a6lpzmpd5yr3xg/include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/spdlog-1.10.0-xvgujvylicokkeotqfa5ae72cvz26bgi/include -isystem /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/fmt-9.1.0-wzitn7apw7mtivdkrb4udfu6fmtdzdu6/include -O2 -g -DNDEBUG -fPIC   -g -Wall -Wextra -pthread -std=gnu++14 -MD -MT src/CMakeFiles/bedrock-client.dir/AsyncRequest.cpp.o -MF CMakeFiles/bedrock-client.dir/AsyncRequest.cpp.o.d -o CMakeFiles/bedrock-client.dir/AsyncRequest.cpp.o -c /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-src/src/AsyncRequest.cpp
In file included from /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/mochi-thallium-0.10.1-ux2i6ha3jhuw5b3tcuhgrqtwvz374mun/include/thallium.hpp:4:0,
                 from /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-src/src/AsyncRequestImpl.hpp:10,
                 from /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-src/src/AsyncRequest.cpp:8:
/pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/mochi-thallium-0.10.1-ux2i6ha3jhuw5b3tcuhgrqtwvz374mun/include/thallium/engine.hpp: In function 'void thallium::_wrapper_for_thallium_generic_rpc(hg_handle_t)':
/pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/mochi-thallium-0.10.1-ux2i6ha3jhuw5b3tcuhgrqtwvz374mun/include/thallium/engine.hpp:1173:8: sorry, unimplemented: non-trivial designated initializers not supported
 inline __MARGO_INTERNAL_RPC_WRAPPER(thallium_generic_rpc)
        ^
In file included from /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/mochi-thallium-0.10.1-ux2i6ha3jhuw5b3tcuhgrqtwvz374mun/include/thallium.hpp:4:0,
                 from /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-src/src/../include/bedrock/Client.hpp:10,
                 from /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-src/src/../include/bedrock/ServiceHandle.hpp:9,
                 from /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-src/src/ServiceHandle.cpp:6:
/pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/mochi-thallium-0.10.1-ux2i6ha3jhuw5b3tcuhgrqtwvz374mun/include/thallium/engine.hpp: In function 'void thallium::_wrapper_for_thallium_generic_rpc(hg_handle_t)':
/pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/mochi-thallium-0.10.1-ux2i6ha3jhuw5b3tcuhgrqtwvz374mun/include/thallium/engine.hpp:1173:8: sorry, unimplemented: non-trivial designated initializers not supported
 inline __MARGO_INTERNAL_RPC_WRAPPER(thallium_generic_rpc)
        ^
In file included from /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/mochi-thallium-0.10.1-ux2i6ha3jhuw5b3tcuhgrqtwvz374mun/include/thallium.hpp:4:0,
                 from /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-src/src/../include/bedrock/ServiceHandle.hpp:14,
                 from /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-src/src/../include/bedrock/Client.hpp:9,
                 from /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-src/src/Client.cpp:7:
/pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/mochi-thallium-0.10.1-ux2i6ha3jhuw5b3tcuhgrqtwvz374mun/include/thallium/engine.hpp: In function 'void thallium::_wrapper_for_thallium_generic_rpc(hg_handle_t)':
/pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/mochi-thallium-0.10.1-ux2i6ha3jhuw5b3tcuhgrqtwvz374mun/include/thallium/engine.hpp:1173:8: sorry, unimplemented: non-trivial designated initializers not supported
 inline __MARGO_INTERNAL_RPC_WRAPPER(thallium_generic_rpc)
        ^
make[2]: *** [src/CMakeFiles/bedrock-client.dir/build.make:107: src/CMakeFiles/bedrock-client.dir/AsyncRequest.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [src/CMakeFiles/bedrock-client.dir/build.make:79: src/CMakeFiles/bedrock-client.dir/Client.cpp.o] Error 1
make[2]: *** [src/CMakeFiles/bedrock-client.dir/build.make:93: src/CMakeFiles/bedrock-client.dir/ServiceHandle.cpp.o] Error 1
make[2]: Leaving directory '/tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o'
make[1]: *** [CMakeFiles/Makefile2:257: src/CMakeFiles/bedrock-client.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 20%] Linking C static library libjx9.a
cd /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o/src && /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/cmake-3.24.3-vkdicyfnbvvdbhghumodnsw43jty6zob/bin/cmake -P CMakeFiles/jx9.dir/cmake_clean_target.cmake
cd /tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o/src && /pfs/lustrep1/users/dewiy/spack/opt/spack/linux-sles15-zen/gcc-7.5.0/cmake-3.24.3-vkdicyfnbvvdbhghumodnsw43jty6zob/bin/cmake -E cmake_link_script CMakeFiles/jx9.dir/link.txt --verbose=1
/opt/cray/pe/cce/14.0.2/binutils/x86_64/x86_64-pc-linux-gnu/bin/ar qc libjx9.a CMakeFiles/jx9.dir/jx9/jx9.c.o
/opt/cray/pe/cce/14.0.2/binutils/x86_64/x86_64-pc-linux-gnu/bin/ranlib libjx9.a
make[2]: Leaving directory '/tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o'
[ 20%] Built target jx9
make[1]: Leaving directory '/tmp/dewiy/spack-stage/spack-stage-mochi-bedrock-0.5.2-6feie4o4btuxomx4gfgnm253scb67jy2/spack-build-6feie4o'
make: *** [Makefile:149: all] Error 2

**Comment by mdorier** [comment]
I honestly don't know what the problem is. The macro it references does not have designated initializers. Besides, if gcc finds a problem inside a macro, it will generally expand the macro to show more precisely where the problem is. Here it does not. I should try myself with gcc 7.5.0 but that means finding a platform that has this compiler, and it's hard to come by nowadays. In the meantime, I would suggest trying to use a more recent version of gcc.

**Comment by mdorier** [comment]
I was able to reproduce the problem with gcc 7.3.0. I'll look into it further.

**Comment by dyokelson** [comment]
That did work, thank you!!!

**Comment by mdorier** [comment]
Thanks, I have released version 0.13.1 of margo. It's available in mochi-spack-packages already and it should be available upstream in spack soon as well.
